### ISSUE TITLE
[Bug]: Fatal Exception: java.util.ConcurrentModificationException

### ISSUE THREAD

**Issue by ajaybirla-fabzen** [OneSignal/OneSignal-Unity-SDK#682]
### What happened?

The application crashes whenever it receives a notification when it is in the background.

### Steps to reproduce?

```Markdown
1. Launch the application.
2. Minimize the application.
3. Send a notification to the application.
4. Click on the notification.
5. Observe the application crash.
```

### What did you expect to happen?

I expect the app to not crash whenever it receives a notification in the background.

### Unity version

2021.3.17f1

### OneSignal Unity SDK version

5.0.5

### Platform

Android

### Relevant log output

```Shell
Fatal Exception: java.util.ConcurrentModificationException:
       at java.util.LinkedHashMap$LinkedHashIterator.nextNode(LinkedHashMap.java:757)
       at java.util.LinkedHashMap$LinkedEntryIterator.next(LinkedHashMap.java:790)
       at java.util.LinkedHashMap$LinkedEntryIterator.next(LinkedHashMap.java:790)
       at com.onesignal.common.modeling.Model.toJSON(Model.kt:312)
       at com.onesignal.common.modeling.ModelStore.persist(ModelStore.kt:141)
       at com.onesignal.common.modeling.SingletonModelStore.replace(SingletonModelStore.kt:36)
       at com.onesignal.core.internal.config.impl.ConfigModelStoreListener$fetchParams$1.invokeSuspend(ConfigModelStoreListener.kt:101)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
       at kotlinx.coroutines.internal.ScopeCoroutine.afterResume(Scopes.kt:33)
       at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:102)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
       at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
       at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284)
       at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
       at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
       at kotlinx.coroutines.BuildersKt.runBlocking(:1)
       at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
       at kotlinx.coroutines.BuildersKt.runBlocking$default(:1)
       at com.onesignal.common.threading.ThreadUtilsKt$suspendifyOnThread$1.invoke(ThreadUtils.kt:60)
       at com.onesignal.common.threading.ThreadUtilsKt$suspendifyOnThread$1.invoke(ThreadUtils.kt:59)
       at kotlin.concurrent.ThreadsKt$thread$thread$1.run(Thread.kt:30)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jinliu9508** [comment]
@ajaybirla-fabzen, thank you for providing the log output! We've acknowledged this issue and plan to address it in the next release.

**Comment by shepherd-l** [comment]
Thank you for your patience, this has been addressed in the [5.0.6 release](https://github.com/OneSignal/OneSignal-Unity-SDK/releases/tag/5.0.6). Let us know if you still see this issue
